test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -708,6 +708,55 @@ class ExtProcIntegrationTest : public HttpIntegrationTest,
     verifyDownstreamResponse(*response, 200);
   }
 
+  IntegrationStreamDecoderPtr initAndSendDataMxnMode(absl::string_view body_sent,
+                                                     bool end_of_stream) {
+    config_helper_.setBufferLimits(1024, 1024);
+    proto_config_.mutable_processing_mode()->set_request_header_mode(ProcessingMode::SEND);
+    proto_config_.mutable_processing_mode()->set_request_body_mode(
+        ProcessingMode::BIDIRECTIONAL_STREAMED);
+    proto_config_.mutable_processing_mode()->set_request_trailer_mode(ProcessingMode::SEND);
+    proto_config_.mutable_processing_mode()->set_response_header_mode(ProcessingMode::SKIP);
+
+    initializeConfig();
+    HttpIntegrationTest::initialize();
+    codec_client_ = makeHttpConnection(lookupPort("http"));
+    Http::TestRequestHeaderMapImpl default_headers;
+    HttpTestUtility::addDefaultHeaders(default_headers);
+
+    auto encoder_decoder = codec_client_->startRequest(default_headers);
+    request_encoder_ = &encoder_decoder.first;
+    auto response = std::move(encoder_decoder.second);
+    codec_client_->sendData(*request_encoder_, body_sent, end_of_stream);
+    return response;
+  }
+
+  void serverReceiveHeaderMxn(ProcessingRequest& header_request) {
+    ASSERT_TRUE(grpc_upstreams_[0]->waitForHttpConnection(*dispatcher_, processor_connection_));
+    ASSERT_TRUE(processor_connection_->waitForNewStream(*dispatcher_, processor_stream_));
+    ASSERT_TRUE(processor_stream_->waitForGrpcMessage(*dispatcher_, header_request));
+    ASSERT_TRUE(header_request.has_request_headers());
```

Comment:
nit: having these be ASSERT will not early-exit the calling function. You can use AssertionResult, and use ASSERT_TRUE at call-site.

Reply:
hmm, need some further understanding on the difference.  Could you please elaborate what's the right way to do it? 
BTW, this kind of pattern is widely used in this test file. Maybe we can have a separate cleanup PR to deal with this?
https://github.com/envoyproxy/envoy/blob/59e8fadde8ade3444796bf13fbda02d2628b3d4c/test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc#L277